lms/djangoapps/badges/api/views.py

```diff
@@ -137,3 +143,60 @@ def get_queryset(self):
                 badge_class__issuing_component=self.request.query_params.get('issuing_component', '')
             )
         return queryset
+
+
+class LeaderboardView(generics.ListAPIView):
+    """
+    Leaderboard List API View
+    """
+    serializer_class = UserLeaderboardSerializer
+
+    def get_queryset(self):
+        """
+        leaderboard queryset
+        """
+        leaderboard_conf = LeaderboardConfiguration.current()
+
+        if leaderboard_conf and leaderboard_conf.enabled:
+            course_badge_score = leaderboard_conf.course_badge_score
+            event_badge_score = leaderboard_conf.event_badge_score
+        else:
+            course_badge_score = LeaderboardConfiguration.COURSE_BADGE_SCORE
+            event_badge_score = LeaderboardConfiguration.EVENT_BADGE_SCORE
+
+        leaderboard_data = (
```

Comment:
This seems very compute intensive query it might not work with large dataset. can we pre compute these on badge generation events to avoid this query scanning complete badge table?